tests/promises/test_concurrent_future.py

```
@@ -0,0 +1,423 @@
+        - Promises with start_soon=False only raise when awaited for directly
+    """
+
+    coro_call_count = 0
+
+    # Create a Promise
+    if start_soon is None:
+        # `start_soon=None` in our test means that we want to create a prefilled promise with exception
+        promise = Promise(prefill_exception=ValueError("Test error from Promise!"))
+    else:
+
+        async def failing_coro():
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `failing_coro`

Add return type annotation: `NoReturn`

(ANN202)